src/overlords/offense/roomPoisoner.ts

```diff
@@ -0,0 +1,159 @@
+import {log} from '../../console/log';
+import {Roles, Setups} from '../../creepSetups/setups';
+import {Pathing} from '../../movement/Pathing';
+import {DirectivePoisonRoom} from '../../directives/offense/poisonRoom';
+import {OverlordPriority} from '../../priorities/priorities_overlords';
+import {profile} from '../../profiler/decorator';
+import {Tasks} from '../../tasks/Tasks';
+import {Zerg} from '../../zerg/Zerg';
+import {MY_USERNAME} from '../../~settings';
+import {Overlord} from '../Overlord';
+import {OvermindConsole} from '../../console/Console';
+import {ColonyMemory} from '../../Colony';
+
+/**
+ * Spawn roomPoisoner - upgrqde controller to lvl2, wall in controller then sources.
+ */
+@profile
+export class RoomPoisonerOverlord extends Overlord {
+
+	roomPoisoners: Zerg[];
+	antiControllers: Zerg[]; //claim and counter reserve if required
+
+    controllerWallSites: ConstructionSite[] | undefined;
+    sourcesWallSites: ConstructionSite[] | undefined;
+
+	constructor(directive: DirectivePoisonRoom, priority = OverlordPriority.offense.roomPoisoner) {
+		super(directive, 'contaminate', priority);
+		this.roomPoisoners = this.zerg(Roles.roomPoisoner);
+		this.antiControllers = this.zerg(Roles.claim);
+
+		this.controllerWallSites = (this.room && this.room.controller) ? _.filter(this.room.constructionSites,
```

Comment:
you can do 

```
if (this.room && this.room.controller) {
    this.controllerWallSites = ...
}
```

which should infer the types so you don't need to do ugly stuff like `this.room!.controller!.pos`

Reply:
have tried it out, did not resolve it 100%, check alternative